mmv1/third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -395,6 +445,33 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 	}
 }
 
+<% unless version == 'ga' -%>
+func resourceComputeSecurityPolicyRulePreconfiguredWafConfigExclusionFieldParamsSchema(description string) *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"operator": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					Default:      "EQUALS",
```

Comment:
Question: Is this the fixed default value in the API? If so, is there a documentation about this?
Same for the "value" field below.

Reply:
The [API documentation](https://cloud.google.com/compute/docs/reference/rest/beta/securityPolicies) doesn't give much simply specifying that op is an enum and val a string. The gcloud documentation for [add-preconfig-waf-exclusion](https://cloud.google.com/sdk/gcloud/reference/beta/compute/security-policies/rules/add-preconfig-waf-exclusion) includes the following additional detail:

> A field value must be given if the field operator is not EQUALS_ANY, and cannot be given if the field operator is EQUALS_ANY.

So to answer the question: No, an op of `EQUALS` is not a fixed default in the API (and I believe you'll get an error if you don't specify the op), but seemed to be the sensible default since operator must be set when specifying a value and it seems rather idiomatic for value to be an exact match when the operator is not explicitly defined in HCL. If you'd rather change operator to a required field with no default, I'd be ok with that.

The default of `""` for the value is effectively a no-op there to support the requirement that value not be set when specifying an operator of EQUALS_ANY. If value is not specified and operator is anything other than EQUALS_ANY, the API will return the following error:

```
Error updating SecurityPolicy "ptlabs-waf-policy": googleapi: Error 400: Invalid value for field 'resource.preconfiguredWafConfig.exclusions[0].requestQueryParamsToExclude[0].val': ''. The field value must be non-empty if the field operator is not EQUALS_ANY., invalid
```

So value must be an optional field on the block, allowing for example (where the default value of `""` is silently ignored and does nothing):

```hcl
              request_cookie {
                  operator = "EQUALS_ANY"
              }
```

Reply:
Thanks for the clear explanation! 
In this case, I would recommend that we set `operator` as a required field (it will also prevent users from specifying an empty block which sometimes ends up causing problems in Terraform) and set `value` as an optional field without any default values, and the API side throw the error if users do not specify the `value` field when `operator` is not `EQUALS_ANY`.